src/circuit/circuit_frame.rs

```diff
@@ -2448,10 +2450,29 @@ fn apply_continuation<F: LurkField, CS: ConstraintSystem<F>>(
             &g.op2_cons_tag,
         )?;
 
+        let arg1_is_char = alloc_equal(
+            &mut cs.namespace(|| "arg1_is_char"),
+            arg1.tag(),
+            &g.char_tag,
+        )?;
+        let arg2_is_str = alloc_equal(&mut cs.namespace(|| "arg2_is_str"), arg2.tag(), &g.str_tag)?;
+        let both_args_are_str = Boolean::and(
+            &mut cs.namespace(|| "both_args_are_str"),
```

Comment:
This an extremely bad variable name/namespace. You've established that one arg is a Char, and the other is a Str. I don't think you should describe this as `both_args_are_str`.

Reply:
Solved